HarClient.Tests/Validation/HarAssert.cs

```diff
@@ -0,0 +1,251 @@
+            if (JAssert.HasOptionalObjectProperty(log, "browser") is JObject browser)
+                AssertIsValidCreatorBrowser(browser);
+            var pageIds = AssertValidPages(
+                JAssert.HasOptionalArrayProperty(log, "pages"));
+            AssertValidEntries(
+                JAssert.HasArrayProperty(log, "entries"), pageIds);
+            _ = HasOptionalComment(log);
+
+            // Log the full JSON to the standard output; we can collect this from the test explorer
+            // and put it into the online HAR validator
+            Console.WriteLine(log.Root.ToString());
+        }
+        static void AssertIsValidCreatorBrowser(JObject creator)
+        {
+            _ = JAssert.HasStringProperty(creator, "name");
+            _ = JAssert.HasStringProperty(creator, "version");
+            _ = HasOptionalComment(creator);
+        }
+        static IReadOnlyCollection<string> AssertValidPages(JArray? pages)
+        {
+            if (pages is null)
+                return Array.Empty<string>();
+            var pageIds = new HashSet<string>();
+            foreach (var member in pages)
+            {
+                Assert.IsTrue(
+                    pageIds.Add(AssertIsValidPage(JAssert.IsObject(member))),
+                    "Pages collection contained duplicate IDs");
+            }
+            return pageIds;
+
+            static string AssertIsValidPage(JObject page)
+            {
+                StringAssert.Matches(
+                    JAssert.HasStringProperty(page, "startedDateTime"),
+                    isoDatetime);
+                var pageId = JAssert.HasStringProperty(page, "id");
+                _ = JAssert.HasStringProperty(page, "title");
+                AssertIsPageTiming(JAssert.HasObjectProperty(page, "pageTimings"));
+                _ = HasOptionalComment(page);
+                return pageId;
+            }
+            static void AssertIsPageTiming(JObject pageTiming)
+            {
+                AssertHasOptionalNumberOrMinusOne(pageTiming, "onContentLoad");
+                AssertHasOptionalNumberOrMinusOne(pageTiming, "onLoad");
+                _ = HasOptionalComment(pageTiming);
+            }
+        }
+        static void AssertValidEntries(JArray entries, IReadOnlyCollection<string> validRefs)
+        {
+            foreach (var member in entries)
+            {
+                var pageref = AssertIsValidEntry(JAssert.IsObject(member));
+                if (pageref != null)
+                    Assert.IsTrue(validRefs.Contains(pageref), "Entry refers to undefined page {0}", pageref);
+            }
+
+            static string? AssertIsValidEntry(JObject entry)
+            {
+                StringAssert.Matches(
+                    JAssert.HasStringProperty(entry, "startedDateTime"),
+                    isoDatetime);
+                var time = JAssert.HasNumericProperty(entry, "time").Value<double>();
```

Comment:
Suggestion: provide another level of "consistency checking" against the `HarSharp.Har` object which can validate that the `time` is consistent with `timings`.

Reply:
Also potentially worthwhile "promoting" entry/pageref into the data model

Reply:
YAGNI.